qlib/TableMapper.qm

```diff
@@ -880,8 +993,15 @@ on_error table_mapper.rollback();
 
             # get return value
             hash rv = hbuf + rh;
-            # clear buffer but keep hash keys
-            map hbuf.$1 = (), hbuf.keyIterator();
+            if (exists rec_list_size) {
```

Comment:
as agreed - hbuf should always contain hash of lists - the problem is with "constant mappers", i.e. if you have a mapper with only constant values. Which is not forbidden. And causing another strange situations anyway - I was also about to forbid constant mappers at all, but let's keep this option for now. I'll address this as part of hbuf changes...